app/src/shared/web3/web3Poller.js

```diff
@@ -0,0 +1,192 @@
+        if (this.ethereumNetworkPollTimeout) {
+            clearTimeout(this.ethereumNetworkPollTimeout)
+            this.ethereumNetworkPollTimeout = undefined
+        }
+    }
+
+    clearPendingTransactionsPoll = () => {
+        if (this.pendingTransactionsPollTimeout) {
+            clearTimeout(this.pendingTransactionsPollTimeout)
+            this.pendingTransactionsPollTimeout = undefined
+        }
+    }
+
+    fetchWeb3Account = () => {
+        this.web3.getDefaultAccount()
+            .then((account) => {
+                this.handleAccount(account)
+                // needed to avoid warnings about creating promise inside a handler
+                // if any other web3 actions are dispatched.
+                return Promise.resolve()
+            }, (err) => {
+                if (this.account !== null) {
+                    this.account = null
+                    this.emitter.emit(events.ACCOUNT_ERROR, err)
+                }
+            })
+    }
+
+    handleAccount = (account: string) => {
+        const next = account
+
+        const didChange = !!(this.account && next && !areAddressesEqual(this.account, next))
+        const didDefine = !!(!this.account && next)
+
+        // Check current provider so that account event is not sent prematurely
+        // (ie. wait for user to approve access to Metamask)
+        if (this.web3.currentProvider !== null && (didDefine || didChange)) {
+            this.account = next
+            this.emitter.emit(events.ACCOUNT, next)
+        }
+    }
+
+    fetchChosenEthereumNetwork = () => {
+        this.web3.getEthereumNetwork()
+            .then((network) => {
+                this.handleNetwork(network.toString())
+            }, () => {
+                this.web3.getDefaultAccount()
+                    .then(() => Promise.resolve(), (err) => {
+                        if (this.account !== null) {
+                            this.emitter.emit(events.NETWORK_ERROR, err)
+                        }
+                    })
+            })
+    }
+
+    handleNetwork = (network: NumberString) => {
+        const next = network
+        const didChange = this.networkId && next && this.networkId !== next
+        const didDefine = !this.networkId && next
+        if (didDefine || didChange) {
+            this.networkId = next
+            this.emitter.emit(events.NETWORK, next)
+        }
+    }
+
+    pollPendingTransactions = () => {
+        this.handlePendingTransactions()
+        this.clearPendingTransactionsPoll()
+        this.pendingTransactionsPollTimeout = setTimeout(this.pollPendingTransactions, FIVE_SECONDS)
+    }
+
+    handlePendingTransactions = () => {
+        const web3 = getPublicWeb3()
+        Object.keys(getTransactionsFromSessionStorage())
+            .forEach((txHash) => {
+                // Get current number of confirmations and compare it with sought-for value
+                hasTransactionCompleted(txHash)
+                    .then((completed) => {
```

Comment:
Lots of uncatchable async going on in here. Whenever you create a promise, you should always return it otherwise nothing can possible handle any errors that occur except the global event handler.

i.e. this should probably be something like:
```js
return Promise.all(Object.keys(getTransactionsFromSessionStorage()).map((txHash) => {
  return hasTransactionCompleted(txHash)
    .then((completed) => {
      if (completed) {
        return web3.eth.getTransactionReceipt(txHash) // always return created promises
          .then((receipt) => {
             if (receipt) {
               // …
             }
          })
    })
}))

```

Reply:
or using async/await
```js
return Promise.all(Object.keys(getTransactionsFromSessionStorage()).map(async (txHash) => {
    const completed = await hasTransactionCompleted(txHash)
    if (completed) {
        const receipt = await web3.eth.getTransactionReceipt(txHash)
        if (receipt) {
            // …
        }
    }
})
```

Reply:
This is a lot cleaner, I used the latter option :)